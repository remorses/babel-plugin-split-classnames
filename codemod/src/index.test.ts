import path from 'path'
import fs from 'fs'

import { transformer, DEFAULT_JSC_OPTIONS, splitClassNames } from './transformer'
import { applyTransform } from 'jscodeshift/dist/testUtils'

for (let testName of fs.readdirSync(path.join(__dirname, 'examples'))) {
    const testFile = path.resolve(__dirname, 'examples', testName)
    const source = fs.readFileSync(testFile, 'utf8').toString()
    test(testName, () => {
        const res = applyTransform(
            transformer,
            DEFAULT_JSC_OPTIONS,
            {
                source,
            },
            DEFAULT_JSC_OPTIONS,
        )
        if (process.env.DEBUG) {
            console.log()
            console.log(res)
            console.log()
        }
        expect(res).toMatchSnapshot(testName)
    })
}

test('splitClassNames keeps all classes', () => {
    const N = 8
    const expectedLen = 'something1'.length * N + (N - 1) * 1
    const cls = new Array(N)
        .fill('')
        .map((_, i) => `something${i}`)
        .join(' ')
    // console.log(cls)
    const res = splitClassNames(cls)
    // console.log(res)
    expect(res.join(' ').length).toBe(expectedLen)
})

test('splitClassNames makes small groups', () => {
    const N = 8
    const expectedLen = 'something1'.length * N + (N - 1) * 1
    const cls = new Array(N)
        .fill('')
        .map((_, i) => `something${i}`)
        .join(' ')
    // console.log(cls)
    const res = splitClassNames(cls, 'something1'.length)
    expect(res.length).toBe(N)
    expect(res.join(' ').length).toBe(expectedLen)
})
test('splitClassNames sorts classes for tailwind', () => {
    const cls =
        'w-1 w-[700px] hover:h-[800px] text-gray-800 hover:text-gray-800 dark:w-5 h-xl absolute relative disabled:w-8 text-white text-black'
    // console.log(cls)
    const res = splitClassNames(cls, 60)

    expect(JSON.stringify(res)).toBe(
        JSON.stringify([
            'w-1 text-gray-800 h-xl absolute relative text-white text-black w-[700px]',
            'disabled:w-8 dark:w-5 hover:text-gray-800 hover:h-[800px]',
        ]),
    )
})
