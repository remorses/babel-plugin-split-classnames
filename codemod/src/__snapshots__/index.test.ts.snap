// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`existingCall.jsx: existingCall.jsx 1`] = `
"import clsx from \\"classnames\\";
function Component() {
    return (
        <Fragment>
            <p
                className={classnames(
                    \\"block w-full py-2 mt-8 text-sm font-semibold text-center text-white\\",
                    \\"bg-gray-900 border border-black rounded-md w-[340px] disabled:opacity-60\\",
                    \\"hover:cursor-pointer disabled:bg-gray-900 hover:bg-gray-700 sm:h-[60em]\\"
                )}
            />
        </Fragment>
    );
}"
`;

exports[`readme.jsx: readme.jsx 1`] = `
"import clsx from \\"classnames\\";
function Component() {
    return (
        <Fragment>
            <p className={clsx(
                \\"block w-full py-2 mt-8 text-sm font-semibold text-center text-white\\",
                \\"bg-gray-900 border border-black rounded-md w-[340px] disabled:opacity-60\\",
                \\"sm:h-[60em] hover:bg-gray-700 disabled:bg-gray-900 hover:cursor-pointer\\"
            )} />
        </Fragment>
    );
}"
`;

exports[`simple.jsx: simple.jsx 1`] = `
"import React from 'react'

import clsx from \\"classnames\\";

export default function Component() {
    return (
        <div>
            <p className='ciao'></p>
            <p
                className={clsx(
                    \\"block w-full py-2 mt-8 text-sm font-semibold text-center text-white\\",
                    \\"bg-gray-900 border block w-full py-2 mt-8 text-sm font-semibold text-center\\",
                    \\"text-white bg-gray-900 border\\"
                )}
            ></p>
            <p
                className={clsx(
                    \\"block w-full py-2 mt-8 text-sm font-semibold text-center text-white\\",
                    \\"bg-gray-900 border\\",
                    \\"another one\\",
                    true && 'another'
                )}
            ></p>
            <p
                className={clsx(
                    \\"block w-full py-2 mt-8 text-sm font-semibold text-center text-white\\",
                    \\"bg-gray-900 border block w-full py-2 mt-8 text-sm font-semibold text-center\\",
                    \\"text-white bg-gray-900 border\\",
                    x
                )}
            ></p>
            <p className={clsx(cond ? 'literal-class' : '')}></p>
            <p className={clsx(\\"x\\", cond ? 'literal-class' : '', \\"z\\")}></p>

            <p className={clsx(\\"ciao sono un literal\\")}></p>
            <p className='a b c d e f g'></p>
        </div>
    );
}"
`;
